### PR TITLE
(1975) Allow users to edit a profession

### PR DESCRIPTION
**Pull request by Gweaton**
# Changes in this PR

- Adds a new journey for editing a profession, accessible via a button on the new `admin/professions/:slug` page (split out from the public facing page)
- Uses a `Profession`'s `confirmed` state (set to `true` when the `Profession` is completed) to determine whether it is being edited or created for the first time.
- Adds captions to the add / edit journey to inform the user of which journey they're currently in.

### New button on `admin/professions/:profession-slug`
image

### Confirmation of editing

image

### Original answers page

image

### Edit page with new "editing" caption

image

### Check your answers page with Publish Now button

image

### Confirmation

image

### CODE REVIEW

File: src/professions/admin/check-your-answers.controller.ts [hunk 1]
```diff
@@ -54,7 +65,17 @@ export class CheckYourAnswersController {
...
       legislation: draftProfession.legislation,
+      confirmed: Boolean(draftProfession.confirmed),
+      edit: Boolean(edit),
       backLink: backLink(req),
     };
   }
+
+  @Post(':id/check-your-answers')
+  @Permissions(UserPermission.CreateProfession)
+  async edit(@Res() res: Response, @Param('id') id: string): Promise<void> {
```
Comment (pezholio) [inline src/professions/admin/check-your-answers.controller.ts:78]: This feels like it should be an `update` action in the `professions` controller.

File: views/admin/professions/check-your-answers.njk [hunk 2]
```diff
@@ -313,14 +317,26 @@
...
-            {{
-              govukButton({
-                id: "submit-button",
-                type: "Submit",
-                preventDoubleClick: true,
-                text: ("professions.form.button.create" | t)
-              })
-            }}
+            {% if edit %}
+            {% elif confirmed%}
```
Comment (pezholio) [inline views/admin/professions/check-your-answers.njk:321]: Slight nitpick:

```suggestion
            {% elif confirmed %}
```

File: src/professions/admin/viewUtils.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export default class ViewUtils {
```
Comment (pezholio) [inline src/professions/admin/viewUtils.ts:1]: 👍

**Review by pezholio — APPROVED** [review]
Lovely stuff. Just a couple of minor things

File: src/professions/admin/professions.controller.ts [hunk 4]
```diff
@@ -56,6 +59,32 @@ export class ProfessionsController {
...
+      ),
+    );
+
+    const industries = await Promise.all(
+      profession.industries.map(
+        async (industry) => await this.i18Service.translate(industry.name),
+      ),
+    );
+
+    return { profession, nations, industries, backUrl: '' };
```
[the remark this answers is not in the document]
Comment (Gweaton) [reply src/professions/admin/professions.controller.ts:86]: Doh, this was after a rebase... fixing now!

File: views/professions/_profession-details.njk [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+<span class="govuk-caption-l">Placeholder Authority</span>
+
+<h1 class="govuk-heading-xl">{{ profession.name }}</h1>
+
+<h2 class="govuk-heading-m rpr-details__section-title">{{ "professions.show.overview.heading" | t }}</h2>
+
+{% set nationsHtml %}
+    {% for nation in nations %}
+      <p class="govuk-body-m"> {{ nation }}
```
[the remark this answers is not in the document]
Comment (Gweaton) [reply views/professions/_profession-details.njk:11]: Oh, good spot.

File: src/professions/admin/check-your-answers.controller.ts (hunk 1 above)
Comment (Gweaton) [reply src/professions/admin/check-your-answers.controller.ts:78]: It very much does. I didn't even write a test for it here 🤫.

Moving it across.

**Pull request merged**